c_api/blockingconcurrentqueue.cpp

```diff
@@ -0,0 +1,49 @@
+#include "concurrentqueue.h"
+#include "../blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> BCQType, *BCQPtr;
+
+extern "C" {
+	
+int moodycamel_bcq_create(BCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return MOODYCAMEL_BAD_HANDLE;
+	}
+	BCQPtr retval = new BCQType;
```

Comment:
With exceptions off, this can result in `nullptr` which I believe you have a special return code for.